\documentclass[a4paper]{article}
\textwidth 6.75in              % set dimensions before fancyhdr 
\textheight 9.5in
\topmargin -.875in
\oddsidemargin -.125in
\evensidemargin -.125in
 
\usepackage{hyperref}
\usepackage{subfigure}

\hypersetup{
    colorlinks=true, %set true if you want colored links
    linktoc=all,     %set to all if you want both sections and subsections linked
    linkcolor=blue,  %choose some color if you want links to stand out
}

\begin{document}
\title{Low miR-200 expression related drug selection}
\author{Quanhu Sheng, Christopher Pendleton}
\maketitle
\tableofcontents

<<setting, echo=FALSE>>=
opts_chunk$set(tidy=TRUE, fig.align='center', size='footnotesize', comment="", fig.width=5, fig.hight=5)
@
\newpage
\section{Purpose}
The miR-200 family contains miR-200a, miR-200b, miR-200c, miR-141, and miR-429. There are growing evidences to suggest that miR-200 microRNAs are involved in cancer metastasis. The project goal is using IC50 data to select drugs that target cell lines with low miR-200 expression.\\

\section{Selection of miR-200 expression related genes}
Here, TCGA paired miRNAseq/RNAseqV2 data was used to find a gene set in which the gene's expression level is highly correlated to the expression level of miR-200. \\
\subsection{Selection of low and high miR-200 expressed sample groups} \label{sec:mirna}
The lowest 5\% and highest 5\% samples ranked by miR-200 expression values from TCGA miRNAseq data were selected as low and high miR-200 expressed groups. DESeq were used to find different expressed genes between those two groups. \\
In order to find out the tumor unspecific gene list, same percentage samples were selected into both low and high miR-200 expressed groups from each tumor dataset. And, only the datasets containing more than 100 samples were used. \\
Since the OV dataset containing 705 miRNAs only is different to the other datasets containing 1046 miRNAs, the OV dataset was discarded.\\
Finally, we got 4270 candidate samples from 13 tumor datasets (Table ~\ref{tab:one}).

<<tumors, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results='asis', cache=TRUE>>=
source("e:/sqh/programs/r/mirna/chris/common.r")
samplefile<-"TCGA_20130521_samples_no_ov_min100.csv"
tumortable<-read.csv(samplefile)
colnames(tumortable)<-c("Tumor","Sample")
tumortable<-tumortable[order(tumortable$Sample, decreasing=TRUE),]
print(xtable(tumortable, caption = "Tumor samples", label = "tab:one"), include.rownames=FALSE, caption.placement="top")
@

\newpage
Table ~\ref{tab:data} is the illustration of miRNA count data.
<<miRNAdata, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results='asis', cache=TRUE>>=
source("e:/sqh/programs/r/mirna/chris/common.r")
data<-read.csv("TCGA_mirnaseq_20130521_count_no_ov_min100.target.csv", row.names=1)
print(xtable(data[1:5,1:3], caption = "miRNA count data", label = "tab:data"), include.rownames=T, caption.placement="top")
@

Following steps were used to do sample selection.
\begin{enumerate}
  \item Variance stabilizing transformed values of each miRNA were calculated by DESeq and used as expression values in cluster analysis.
  \item 296 miRNAs observed in at least 90\% samples were kept and clustered (Figure ~\ref{fig:mirna296})
    \begin{figure}[!ht]
      \begin{subfigure}{
        \includegraphics[width=.1\linewidth] {H:/shengquanhu/projects/miRNA/TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.target.DESeq.heatmap.color.png}
      }
      \end{subfigure}\hfill
      \begin{subfigure}{
        \includegraphics[width=.8\linewidth]{H:/shengquanhu/projects/miRNA/TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.DESeq.heatmap.png}
      }
      \end{subfigure}\hfill
      \centering
      \caption{The cluster result using all 296 miRNAs. It indicates that the expression pattern of overall miRNAs was tumor related.} \label{fig:mirna296}
    \end{figure}
  \item The cluster result using miR-200 only indicated that miR-200 were constantly low expressed in lgg and kirc (Figure ~\ref{fig:miR200}).
    \begin{figure}[!ht]
      \begin{subfigure}{
        \includegraphics[width=.1\linewidth] {H:/shengquanhu/projects/miRNA/TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.target.DESeq.heatmap.color.png}
      }
      \end{subfigure}\hfill
      \begin{subfigure}{
        \includegraphics[width=.8\linewidth] {H:/shengquanhu/projects/miRNA/TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.target.DESeq.heatmap.png}
      }
      \end{subfigure}\hfill
      \caption{The cluster result using miR-200 only. It indicated that the expression pattern of miR-200 was almost tumor free, except for lgg and kirc.} \label{fig:miR200}
      \centering
    \end{figure}
  \item Samples from lgg and kirc datasets were excluded and 3588 samples kept.
  \item Ranks of all 296 miRNAs in each sample were calculated.
  \item The rank of miR-200 were calculated by the mean rank of five miR-200 miRNAs in each sample.
  \item Samples were ranked by the rank miR-200.
  \item Top 5\% and bottom 5\% samples in the ranked samples of each tumor were selected and combined as low and high miR200 expressed groups, and the corresponding miR-200 expression values were clustered (Figure ~\ref{fig:miR200_lowhigh}).
    \begin{figure}[!ht]
      \centering
        \includegraphics[width=0.9\linewidth]{H:/shengquanhu/projects/miRNA/TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.rank.target.lowhigh_no_igg_kirc.DESeq.heatmap.png}
      \caption{The cluster result using miR-200 in low and high expressed sample groups. It indicates that the low and high groups were perfectly clustered together using miR200 expression values.} \label{fig:miR200_lowhigh}
    \end{figure}
\end{enumerate}

Finally we got 180 low expressed miR-200 samples and 180 high expressed miR-200 samples (Table ~\ref{tab:samples}). Those two sample groups were used in gene differencial expression analysis.
<<samples, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results='asis', cache=TRUE>>=
source("e:/sqh/programs/r/mirna/chris/common.r")
samples<-read.table("TCGA_mirnaseq_20130521_countexpr_no_ov_min100.t.90.rank.target.lowhigh_no_igg_kirc.sample.tsv", sep="\t", header=T, row.names=1)
out<-matrix ( unlist( strsplit(rownames(samples), "_") ), byrow=T, ncol=2)
tumortable<-as.matrix(table(out[ ,1]))
xtab<-data.frame(Tumor=rownames(tumortable), Low=unlist(tumortable[,1]), High=unlist(tumortable[,1]))
print(xtable(xtab, caption = "Tumor samples used in gene selection analysis", label = "tab:samples"), include.rownames=F, caption.placement="top")
@
\newpage
\subsection{Gene selection by differencial expression analysis} \label{sec:geneselection}
The RNAseq data corresponding to low/high expressed miRNA-200 family samples were extracted from TCGA RNAseqV2 dataset. Genes were selected by following steps:
\begin{enumerate}
  \item Variance stabilizing transformed values of 20501 genes were calculated by DESeq and were used for cluster.
  \item 17651 of 20501 genes observed in at least 180 samples (half of samples).
  \item Differencial expression PValue of each gene was calculated by DESeq using generalized linear model and adjusted by Bonferroni correction
  \item Multiple criteria were used to filter the differencial expressed genes. Total 345 genes passed the criteria. 
    \begin{enumerate}
      \item GLM must be converged
      \item $|log_{2}(foldchange)|\geq1.5$
      \item FDR \ensuremath{\le} 0.01
    \end{enumerate}
  \item Clustering of the significant differencial expressed genes in 360 samples (Figure ~\ref{fig:genes})
    \begin{figure}[!ht]
      \centering
        \includegraphics[width=0.8\linewidth]{H:/shengquanhu/projects/miRNA/TCGA_rnaseqv2_20130521_count_no_ov_min100_no_igg_kirc_miRNAlowhigh.DESeq_glm.sig_abs(fold)_1.5.csv.heatmap.png}
      \caption{The cluster result using 345 genes. It indicated that the low and high expressed miR-200 family groups were roughly clustered together. [How to get a measurement?]} \label{fig:genes}
    \end{figure}
\end{enumerate}

\newpage
\section{Low miR-200 expression group prediction model development}
The NCI60 dataset includes both miRNA and microarray data from 60 tumor cell lines. The miRNA data was download from \url{http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE26375} and the mRNA data was downloaded from \url{http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5846}.\\
The NCI60 dataset was used to validate if the gene set we selected in section \ref{sec:geneselection} can be used to group new samples into low and high miR-200 expressed groups.
\subsection{Validation of high correlation between the selected gene set expression and the miR-200 expression}
\begin{enumerate}
  \item Ranks of all 296 miRNAs (section \ref{sec:mirna}) in each sample were calculated .
  \item The rank of miR-200 in each sample were calculated as section \ref{sec:mirna}.
  \item Samples were ranked by the rank of miR-200 .
  \item 265 genes in NCI60 microarray data overlapped with the gene set (345 genes, section \ref{sec:geneselection}) were kept and clustered (Figure \ref{fig:nci60microarray}).
    \begin{figure}[!ht]
      \centering
        \includegraphics[width=0.8\linewidth]{H:/shengquanhu/projects/miRNA/NCI60_microarray_CEL_filtered.tsv.png}
      \caption{The cluster result using 265 gene expression values in NCI60 microarray data.} \label{fig:nci60microarray}
    \end{figure}
\end{enumerate}
\subsection{Developing a model to predict the group (low/middle/high miR-200 expressed) that an new microarray data will belong to.}
in pregress...

\section{Selection of drugs targeting low miR-200 expressed cell lines}
The IC50 data was dowload from \url{http://www.cancerrxgene.org/downloads/}. IC\_50 means half maximal inhibitory (50\%) drug concentrations (natural log microMolar). The data includes cell line drug sensitivity data, mutations in cancer genes and tissue type (Table \ref{tab:ic50}). It includes 714 cell lines and 138 drugs. \\

<<ic50, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results='asis', cache=TRUE>>=
source("e:/sqh/programs/r/mirna/chris/common.r")
ic50<-read.csv("gdsc_manova_input_w2.csv", head=T)
ic50<-ic50[-1,]
ic50slim<-ic50[1:4,c(1:3,77:78)]
print(xtable(ic50slim, caption = "IC50 data", label = "tab:ic50"), include.rownames=F, caption.placement="top")
@
\subsection{Using model to select low miR-200 expressed cell lines}
pending...
\subsection{Selection of drugs targeting selected cell lines }
pending...

\end{document}