\documentclass[a4paper]{article}
\textwidth 6.75in  
\textheight 9.5in
\topmargin -.875in
\oddsidemargin -.125in
\evensidemargin -.125in

\usepackage[nogin]{Sweave}
\usepackage{hyperref}
\usepackage{pdfpages}

\hypersetup{
    colorlinks=true, %set true if you want colored links
    linktoc=all,     %set to all if you want both sections and subsections linked
    linkcolor=blue,  %choose some color if you want links to stand out
}

\begin{document}
\title{VANGARD00095}
\author{Quanhu Sheng}
\maketitle
\tableofcontents

<<setting, echo=FALSE>>=
opts_chunk$set(tidy=TRUE, fig.align='center', size='footnotesize', comment="", fig.width=5, fig.hight=5)
@

\newpage
\section{Purpose}
Detecting somatic single-nucleotide variants and insertions/deletions from whole exome sequence data generated by VANTAGE on a cancer patient's tumor (frozen tissue from a left breast mass) and her matched frozen blood.\\

\section{Dataset}
\begin{table}[ht]
\centering
\caption{Sample information} 
\label{table:one}
\begin{tabular}{|c|c|c|c|c|c|}
\hline 
Name & Type & File 1 & Size 1 & File 2 & Size 2\tabularnewline
\hline 
\hline 
2562-JP-1 & Normal & 2562-JP-1\_1.fastq.gz & 2903813012 & 2562-JP-1\_2.fastq.gz & 2993732713\tabularnewline
\hline 
2562-JP-2 & Tumor & 2562-JP-2\_1.fastq.gz & 3419197916 & 2562-JP-2\_2.fastq.gz & 3513169391\tabularnewline
\hline 
\end{tabular}
\end{table}

\section{Analysis}
\subsection{Database}
Human genome version 19
\subsection{Raw file quality control}
FastQC, v0.10.1
\subsection{Data pre-processing}
\subsubsection{Alignment}
bwa, v0.7.4-r385
\subsubsection{Realignment}
GATK, v2.4.9, RealignerTargetCreator + IndelRealigner
\subsubsection{Recalibration}
GATK, v2.4.9, BaseRecalibrator + PrintReads
\subsubsection{Remove duplication}
Picard, v1.9.0, MarkDuplicates
\subsection{Call SNP}
GATK, v2.4.9, UnifiedGenotyper\\
SNP remove criteria (based on \href{http://www.broadinstitute.org/gatk/guide/topic?name=best-practices}{GATK best-practices}): 
$$QD<2.0||MQ<40.0||FS>60.0||HaplotypeScore>13.0||MQRankSum<-12.5||ReadPosRankSum<-8.0$$
Insert/deletion remove criteria (based on \href{http://www.broadinstitute.org/gatk/guide/topic?name=best-practices}{GATK best-practices}): 
$$QD<2.0||ReadPosRankSum<-20.0||InbreedingCoeff<-0.8||FS>200.0$$ 
\subsection{Call somatic mutations}
muTect v1.1.4 [\href{http://www.nature.com/nbt/journal/v31/n3/full/nbt.2514.html}{Nat Biotechnology (2013).doi:10.1038/nbt.2514}]
\subsection{Annotation}
annovar, v2013jun21

\section{Result}
\subsection{Raw file quality control}
FastQC passed\\
\includegraphics[page=1, scale=0.24]{FastQC}
\includegraphics[page=2, scale=0.24]{FastQC}\\
\includegraphics[page=3, scale=0.24]{FastQC}
\includegraphics[page=4, scale=0.24]{FastQC}\\

\newpage
\subsection{SNPs}
<<snp, echo=FALSE, results='asis'>>=
library(xtable)
for(name in c("indel", "snp")){
  for(index in 1:2){
    file<-paste0("H:/shengquanhu/projects/vangard/VANGARD00095_jennifer_exome/SNPindel/result/2562-JP-", index, "/2562-JP-", index, "_", name, "_filtered.pass.annovar.hg19_multianno.csv")
    data<-read.csv(file)
    tb<-as.matrix(table(data$Func.refGene))
    tb<-tb[order(tb[,1], decreasing=TRUE),]
    stb<-data.frame(Type=names(tb), Freq=unlist(tb))
    print(xtable(stb, caption = paste0("2562-JP-", index, " ", name, " [", sum(tb), "]"),  label = paste0("tab:", name, index)), include.rownames=F, caption.placement="top")
  }
}
@
\newpage
\subsection{Somatic mutations}
\subsubsection{muTect(375)}
The TP53 somatic mutation was identified as we expected. The brief information of nonsynonymous somatic mutations is illustrated at table ~\ref{tab:four}.  
More detail information can be found at:\\ 2562-JP.somatic.pass.annovar.genome\_summary.csv.xls

<<muTect, echo=FALSE, results='asis'>>=
library(xtable)
data<-read.csv("H:/shengquanhu/projects/vangard/VANGARD00095_jennifer_exome/muTect/result/2562-JP/2562-JP.somatic.pass.annovar.genome_summary.csv", header=FALSE)
data<-data[,c(1:26,36,37)]
colnames(data)<-unlist(data[1,])
colnames(data)[27]<-"Normal"
colnames(data)[28]<-"Tumor"
data<-data[-1,]

functable<-as.data.frame(table(data[,1]))
functable<-functable[functable$Freq > 0,]
colnames(functable)[1]<-"Type"
fod<-order(functable[,2],decreasing=T )
functable<-functable[fod,]
print(xtable(functable, caption = "Func frequencies", label = "tab:two"), include.rownames=FALSE, caption.placement="top")

exontable<-as.data.frame(table(data[data$Func == "exonic",3]))
exontable<-exontable[exontable$Freq > 0,]
colnames(exontable)[1]<-"Type"
fod<-order(exontable[,2],decreasing=T )
exontable<-exontable[fod,]
print(xtable(exontable, caption = "Exonic frequencies", label = "tab:three"), include.rownames=FALSE, caption.placement="top")

nonsym<-data[data$ExonicFunc == "nonsynonymous SNV",]
nonsym<-nonsym[,c(2, 9, 10, 22:28)]
print(xtable(nonsym, caption = "Nonsynonymous SNV", label = "tab:four"), include.rownames=FALSE, caption.placement="top", size="\\tiny")
@

\end{document}